### PR TITLE
Switch from relying directly on Replicache to an interface.

### PR DESCRIPTION
**Pull request by aboodman**
This makes it easier to use with Reflect.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +37,7 @@
     "prettier": "^2.4.1",
     "react": ">=16.0 <18.0",
     "react-dom": ">=16.0 <18.0",
-    "replicache": ">= 10.0.0",
+    "replicache": "10.0.0",
```
Comment (arv) [inline package.json:40]: I would expect this to be `^10.0.0`

**Review by arv — COMMENTED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (aboodman) [reply package.json:40]: Well here's the way I think of it -- we're not actually relying on a particular version . This code is basically do the same thing as the Go pattern of having consumers declare the interfaces for their dependencies, except that in this case the interface is so complex (ReadTransaction) that I'm borrowing the definition in Replicache 10. I could actually pick any version of Replicache that has an interface that is compatible with what useSubscribe is trying to do. I sort of arbitrarily picked 10.

Comment (arv) [reply package.json:40]: I agree.

I was just commenting on locking the version like this. Generally you'd want to use semver.

**Pull request merged**